src/fread.c

```diff
@@ -1971,7 +1971,7 @@ int freadMain(freadMainArgs _args) {
       int thisLineLen = (int)(ch-lineStart);  // ch is now on start of next line so this includes line ending already
       sampleLines++;
       sumLen += thisLineLen;
-      sumLenSq += thisLineLen*thisLineLen;
+      sumLenSq += (double)thisLineLen*thisLineLen; // avoid integer overflow, #6729
```

Comment:
this can be tested without _too_ much pain right? 

IINM we need a line with width ≥46341 in a file that does line sampling.

If you can write such a test in 10 minutes or so please go ahead (i.e. don't slave away too much here)

Reply:
Can it be a test that only fails with UBSan enabled? Even if we overflow here, end up with a negative `sumLenSq` and `sd` equal to `NaN`, later we use `fmax(meanLineLen - 2*sd, minLen)` when calculating `allocnrow`, and `fmax` will return `minLen` if the other argument is `NaN`, so the overflow doesn't spread. Even making `meanLineLen - 2*sd` equal to 0 won't cause any damage because `minLen` is at least 1 (line lengths include the line terminators).

Without a sanitizer, I think this is only noticeable as a wrong initial allocation.

Reply:
Yea, I think we have a number of tests in the suite that similarly only fail in certain settings. As long as the test is not too expensive to run I think it's fine to leave it in there.

Reply:
I confirm that I see the "cannot be represented" message from UBSan in 28e9ed0b4ca704b74718069795050b7532c3c186 but that it's absent in 3d0a592a4e5e4609ca5e084467489f7bd6aacc77.